pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -123,130 +54,57 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 // success response with no actual body content.
 // [1] https://docs.docker.com/registry/spec/api/#existing-layers
 func (pbs *pullthroughBlobStore) ServeBlob(ctx context.Context, w http.ResponseWriter, req *http.Request, dgst digest.Digest) error {
-	store, ok := pbs.digestToStore[dgst.String()]
-	if !ok {
+	if _, err := pbs.BlobStore.Stat(ctx, dgst); err == nil {
 		return pbs.BlobStore.ServeBlob(ctx, w, req, dgst)
 	}
 
+	remoteGetter, found := RemoteBlobGetterFrom(ctx)
+	if !found {
+		err := fmt.Errorf("pullthroughBlobStore.ServeBlob: failed to retrieve remote getter from context")
+		context.GetLogger(ctx).Error(err)
+		return err
```

Comment:
Don't expose implementation details to clients (`pullthroughBlobStore.ServeBlob: `).